node.go

```diff
@@ -96,14 +97,26 @@ func (n *ProtoNode) CidBuilder() cid.Builder {
 }
 
 // SetCidBuilder sets the CID builder if it is non nil, if nil then it
-// is reset to the default value
-func (n *ProtoNode) SetCidBuilder(builder cid.Builder) {
+// is reset to the default value. An error will be returned if the builder
+// is not usable.
+func (n *ProtoNode) SetCidBuilder(builder cid.Builder) error {
```

Comment:
@Jorropo this is the API change that would indicate a semver-minor bump for our <0 state. Adding the `error` here is apparently going to cause trouble. The best case for this being "breaking" was made by @marten-seemann:

> It's a breaking change, strictly speaking. Someone could have saved your function to a variable, and that variable now has a new type 

But regardless, I think we bundle this with #92 for a release and do a semver-minor with that anyway because it's introducing new strictness that wasn't previously there.

Reply:
Nvm I somehow missed that `error` didn't used to be here.